exercises/luhn/src/example.erl

```diff
@@ -34,14 +48,26 @@ checksum([H | ReversedNumber], even, Total) when H >= $5 ->
 
 
 valid(Number) ->
-  checksum(Number) rem 10 == 0.
-
-
-
-create(Number) ->
-  lists:flatten([Number, ($: - (checksum(Number ++ [$0]) rem 10))]).
+  case is_numeric(Number) of
+    true ->
+      case is_min_length(Number) of
```

Comment:
This shouldn't be necessary, since we match on the filtered string below (`Number2 = [_|_] -> ...`) which will only suceed on a string that has at least one character.

Reply:
After this `case` has been removed, it should be possible to remove `is_min_number/1` as well.

Reply:
While the filtered string will only succeed on a string with at least one character, this does not meet the requirement of a luhn number to have a minimum length *excluding* spaces. The `is_min_length/1` tests a minimum length excluding spaces in input, which are part of the test data.

I have made the two other amendments, but should we still revise this `valid` method, or is this acceptable?

Reply:
After you have applied the `lists:filter(fun(C) -> $0 =< C and C =< $9 end, Number)` there will be no space anymore. It will be digits only:

```plain
1> case lists:filter(fun(C) -> ($0 =< C) and (C =< $9) end, "     ") of
1>   [_|_] -> "long enough";
1>   []    -> "too short!!"
1> end.
"too short!!"
```